lib/ansible/modules/notification/rabbitmq_publish.py

```diff
@@ -0,0 +1,182 @@
+  content_type:
+    description:
+      - The content type of the body.
+    default: text/plain
+  durable:
+    description:
+      - Set the queue to be durable.
+    default: False
+    type: bool
+  exclusive:
+    description:
+      - Set the queue to be exclusive.
+    default: False
+    type: bool
+  auto_delete:
+    description:
+      - Set the queue to auto delete.
+    default: False
+    type: bool
+  headers:
+    description:
+      - A dictionary of headers to post with the message.
+    default: {}
+    type: dict
+
+
+requirements: [ pika ]
+notes:
+  - This module requires the pika python library U(https://pika.readthedocs.io/).
+  - Pika is a pure-Python implementation of the AMQP 0-9-1 protocol that tries to stay fairly independent of the underlying network support library.
+author: "John Imison (@Im0)"
+'''
+
+EXAMPLES = '''
+- name: Publish a message to a queue with headers
+  rabbitmq_publish:
+    url: "amqp://guest:guest@192.168.0.32:5672/%2F"
+    queue: 'test'
+    body: "Hello world from ansible module rabitmq_publish"
+    content_type: "text/plain"
+    headers:
+      myHeader: myHeaderValue
+  delegate_to: localhost
+
+
+- name: Publish a file to a queue
+  rabbitmq_publish:
+    url: "amqp://guest:guest@192.168.0.32:5672/%2F"
+    queue: 'images'
+    file: 'path/to/logo.gif'
+  delegate_to: localhost
+
+- name: RabbitMQ auto generated queue
+  rabbitmq_publish:
+    url: "amqp://guest:guest@192.168.0.32:5672/%2F"
+    body: "Hello world random queue from ansible module rabitmq_publish"
+    content_type: "text/plain"
+  delegate_to: localhost
+'''
+
+RETURN = '''
+result:
+  description:
+    - Contains the a message (msg), content type (content_type) and the queue name (queue).
+  returned: success
+  type: dict
+  sample: |
+    'result': { 'content_type': 'text/plain', 'msg': 'Successfully published to queue test', 'queue': 'test' }
+'''
+
+try:
+    import pika
+    HAS_PIKA = True
+except ImportError:
+    HAS_PIKA = False
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.rabbitmq import RabbitClient
+
+
+def main():
+    argument_spec = RabbitClient.rabbitmq_argument_spec()
+    argument_spec.update(
+        exchange=dict(type='str', default=''),
+        routing_key=dict(type='str', required=False),
+        body=dict(type='str', required=False),
+        src=dict(aliases=['file'], type='str', required=False),
+        content_type=dict(default="text/plain", type='str'),
+        durable=dict(default=False, type='bool'),
+        exclusive=dict(default=False, type='bool'),
+        auto_delete=dict(default=False, type='bool'),
+        headers=dict(default={}, type='dict')
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    rabbitmq = RabbitClient(module)
+
+    if module.params['body'] is not None and module.params['src'] is not None:
```

Comment:
`mutually_exclusive` could be used here ([example](https://github.com/ansible/ansible/blob/94eab56d51c8810bbf1ca39015f58f9112e83b07/lib/ansible/modules/system/systemd.py#L323)).